plugins/common/kafka/config.go

```diff
@@ -102,5 +124,27 @@ func (k *Config) SetConfig(config *sarama.Config) error {
 		config.Metadata.Full = *k.MetadataFull
 	}
 
+	if k.MetadataRetryMax != 0 {
+		config.Metadata.Retry.Max = k.MetadataRetryMax
+	}
+
+	if k.MetadataRetryBackoff != 0 {
+		// If config.Metadata.Retry.BackoffFunc is set, sarama ignores
+		// config.Metadata.Retry.Backoff
+		config.Metadata.Retry.Backoff = k.MetadataRetryBackoff
+	}
+
+	switch strings.ToLower(k.MetadataRetryType) {
+	default:
+		return fmt.Errorf("invalid metadata retry type")
+	case "exponential":
+		if k.MetadataRetryBackoff != 0 {
+			config.Metadata.Retry.BackoffFunc = makeBackoffFunc(k.MetadataRetryBackoff, k.MetadataRetryMaxDuration)
+		}
+	case "constant":
+		fallthrough
+	case "":
```

Comment:
Maybe
```suggestion
	case "", "constant":
```
is more clear?